src/main/java/org/jitsi/jicofo/recording/Recorder.java

```diff
@@ -68,7 +76,12 @@ public void init() { }
      */
     public void dispose()
     {
-        xmpp.removePacketHandler(this);
+        if (this.packetProcessor != null)
+        {
+            this.packetProcessor.stop();
```

Comment:
I think first you should remove the listener then stop the processor - to avoid submitting new tasks on stopped executor. I don't remember if that would cause any Exceptions.